### PR TITLE
Update URLs for developer docs

### PR DESCRIPTION
**Pull request by sethobey**
### Context

Dev docs link has changed.

### CODE REVIEW

File: .github/issue_template.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ### Prerequisites
 
 - [ ] Are you running the latest version of our extension?
-- [ ] Have you reviewed our [integration guide](https://www.taxjar.com/guides/integrations/magento2/)?
+- [ ] Have you reviewed our [integration guide](https://developers.taxjar.com/integrations/quides/magento2/)?
```
Comment (jordan-k-johnson) [inline .github/issue_template.md]: Should the link here be 'https://developers.taxjar.com/integrations/guides/magento2/'?

File: Ui/DataProvider/Product/Form/Modifier/TaxTip.php [hunk 2]
```diff
@@ -28,8 +28,8 @@ class TaxTip extends AbstractModifier
     public function modifyMeta(array $meta)
     {
         if (isset($meta['product-details']['children']['container_tax_class_id']['children']['tax_class_id'])) {
-            $url = 'https://www.taxjar.com/guides/integrations/magento2/#product-sales-tax-exemptions';
-            $desc = 'TaxJar requires a product tax class assigned to a TaxJar category in order to exempt products from 
+            $url = \Taxjar\SalesTax\Model\Configuration::TAXJAR_MAGETWO_GUIDE_URL . '/#product-sales-tax-exemptions';
```
Comment (jordan-k-johnson) [inline Ui/DataProvider/Product/Form/Modifier/TaxTip.php]: Should this link go to 'https://developers.taxjar.com/integrations/guides/magento2/#section-product-sales-tax-exemptions' with 'section-' included in the named anchor? 'https://developers.taxjar.com/integrations/guides/magento2/#product-sales-tax-exemptions' doesn't seem to work for me.

**Review by jordan-k-johnson — CHANGES_REQUESTED** [review]

File: .github/issue_template.md (hunk 1 above)
Comment (sethobey) [reply .github/issue_template.md]: 🤦‍♂️ 👍

**Review by jordan-k-johnson — APPROVED** [review]
LGTM

**Review by dallendalton — APPROVED** [review]

**Pull request merged**